# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/sysklogd/module_h_hotfix.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./ksym_mod.c.orig	Fri Nov 24 09:13:46 2000
+++ ./ksym_mod.c	Fri Nov 24 09:13:26 2000
@@ -95,6 +95,13 @@
 #include <linux/time.h>
 #include <linux/module.h>
 #else /* __GLIBC__ */
+
+/* This definition is needed by <linux/module.h>. It's taken from
+   <linux/list.h> where it's only defined if __KERNEL__ is set. - Clifford */
+struct list_head {
+	struct list_head *next, *prev;
+};
+
 #include <linux/module.h>
 extern __off64_t lseek64 __P ((int __fd, __off64_t __offset, int __whence));
 extern int get_kernel_syms __P ((struct kernel_sym *__table));
